heat/core/factories.py

```diff
@@ -434,6 +434,7 @@ def array(
 def asarray(
     obj: Iterable,
     dtype: Optional[Type[datatype]] = None,
+    copy: bool = False,
```

Comment:
According to the array-API standard, `asarray(copy=False)` should return `ValueError` if a copy turns out to be necessary. 
https://data-apis.org/array-api/latest/API_specification/generated/signatures.creation_functions.asarray.html

I propose to set this default to `None` (we won't copy unless necessary), which is basically the default numpy users expect. Then if the users specify `asarray(copy=False)` and copying is necessary, we can return a `ValueError`.

Reply:
I've changed the default and added a check in the event `copy=False`.